frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java

```diff
@@ -161,30 +160,32 @@ public void setTestResponse(EntityModel<Response> testResponse) {
     public UploadImageModel(final Guid limitToStorageDomainId, final DiskImage resumeUploadDisk) {
         if (resumeUploadDisk == null) {
             setDiskModel(new NewDiskModel() {
+                private final boolean isChangeable = limitToStorageDomainId == null;
+
                 @Override
                 public void initialize() {
                     super.initialize();
 
-                    getStorageDomain().setIsChangeable(limitToStorageDomainId == null);
-                    getDataCenter().setIsChangeable(limitToStorageDomainId == null);
+                    getDataCenter().setIsChangeable(isChangeable);
+                    if (!isChangeable) {
+                        initStorageDomain();
```

Comment:
IMHO the name `initStorageDomain()` is not 100% correct and instead of thinking about a better name maybe it would be the best option just to inline the method (IDE can do it without introducing bugs), remove the redundant  "{}" braces around `getStorageDomain().setSelectedItem()` and modify indentation to improve readability.
Suggestion:
```
if (!isChangeable) {
    AsyncDataProvider.getInstance().getStorageDomainById(
            new AsyncQuery<>(storageDomain -> getStorageDomain().setSelectedItem(storageDomain)),
            limitToStorageDomainId);
}
```

Reply:
yeah, or if we set the SDs to a list with a single item then that item is supposed to be selected and we also address the comment below, no?

Reply:
@barpavel done - inlined the method.
@ahadas Since we still need to get the storage domain by its id, so this one is still required.
WDYT?

Reply:
what I meant here is to do: `getStorageDomain().setItems(Collections.singltonList(storageDomain))` - would it work?

Reply:
ACK, cool, seems to work :)